gobblin-restli/gobblin-flow-config-service/gobblin-flow-config-service-server/src/main/java/org/apache/gobblin/service/FlowExecutionResource.java

```diff
@@ -65,8 +66,11 @@ public FlowExecutionResource() {}
    */
   @Override
   public FlowExecution get(ComplexResourceKey<FlowStatusId, EmptyRecord> key) {
-    // this returns null to raise a 404 error if flowStatus is null
-    return convertFlowStatus(getFlowStatusFromGenerator(key, this._flowStatusGenerator));
+    FlowExecution flowExecution = convertFlowStatus(getFlowStatusFromGenerator(key, this._flowStatusGenerator));
+    if (flowExecution == null) {
+      throw new RestLiServiceException(HttpStatus.S_404_NOT_FOUND, "No flow execution found for flowStatusId " + key.getKey());
```

Comment:
When can this happen? Was the wrong flow id provided and user needs to check it, or the flows can get cleaned up/deleted after some period of time? Can it be the problem with permissions that he does not see the flow?

Reply:
It would mean either wrong flow id or the flow status was cleaned up.

Reply:
Can you add it to the message? The point here is to provide detailed and helpful error messages to users, so they don't come back to us with follow up questions, and can troubleshoot themselves.